shuup/default_reports/reports/customer_sales.py

```diff
@@ -30,7 +30,7 @@ class CustomerSalesReport(OrderReportMixin, ShuupReportBase):
 
     def get_objects(self):
         return Contact.objects.filter(
-            customer_orders__in=super(CustomerSalesReport, self).get_objects()
+            customer_orders__in=super(CustomerSalesReport, self).get_objects()[:self.report_limit]
```

Comment:
This limit is being added in the wrong place, please put it after the final filter below:
```py
.filter(order_count__gt=0).order_by("-%s" % self.options["order_by"])[:self.report_limit].values(
    "name", "order_count", "average_sales", "taxless_total", "taxful_total"
)
```